auth/src/main/java/com/firebase/ui/auth/data/remote/TwitterSignInHandler.java

```diff
@@ -0,0 +1,119 @@
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.model.User;
+import com.firebase.ui.auth.viewmodel.idp.ProviderHandlerBase;
+import com.firebase.ui.auth.viewmodel.idp.ProviderHandlerParamsBase;
+import com.firebase.ui.auth.viewmodel.idp.ProvidersHandlerBase;
+import com.google.firebase.auth.TwitterAuthProvider;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.Twitter;
+import com.twitter.sdk.android.core.TwitterAuthConfig;
+import com.twitter.sdk.android.core.TwitterConfig;
+import com.twitter.sdk.android.core.TwitterCore;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.TwitterSession;
+import com.twitter.sdk.android.core.identity.TwitterAuthClient;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class TwitterSignInHandler extends ProviderHandlerBase<TwitterSignInHandler.Params> {
+    private final TwitterAuthClient mClient;
+    private final Callback mCallback = new Callback();
+
+    public TwitterSignInHandler(Application application) {
+        super(application);
+        initialize();
+        mClient = new TwitterAuthClient();
+    }
+
+    public static void signOut() {
+        try {
+            TwitterCore.getInstance();
+        } catch (IllegalStateException e) {
+            initialize();
+        }
+
+        TwitterCore.getInstance().getSessionManager().clearActiveSession();
+    }
+
+    private static void initialize() {
+        Context context = AuthUI.getApplicationContext();
+        TwitterConfig config = new TwitterConfig.Builder(context)
+                .twitterAuthConfig(new TwitterAuthConfig(
+                        context.getString(R.string.twitter_consumer_key),
+                        context.getString(R.string.twitter_consumer_secret)))
+                .build();
+        Twitter.initialize(config);
+    }
+
+    private static IdpResponse createIdpResponse(
+            TwitterSession session, String email, String name, Uri photoUri) {
+        return new IdpResponse.Builder(
+                new User.Builder(TwitterAuthProvider.PROVIDER_ID, email)
+                        .setName(name)
+                        .setPhotoUri(photoUri)
+                        .build())
+                .setToken(session.getAuthToken().token)
+                .setSecret(session.getAuthToken().secret)
+                .build();
+    }
+
+    public TwitterAuthClient getClient() {
+        return mClient;
+    }
+
+    public Callback getCallback() {
+        return mCallback;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        mClient.onActivityResult(requestCode, resultCode, data);
+    }
+
+    private class Callback extends com.twitter.sdk.android.core.Callback<TwitterSession> {
```

Comment:
Does this class have to be called `Callback`?  It conflicts with the `Twitter` class making this unnecessarily hard to read (as you can see)